### PR TITLE
Use entry_points for pool driver discovery

### PR DESCRIPTION
**Pull request by kalkin**
- Add qubes.storage entry point to setup.py
- Removed the old pool driver class loading logic
- Reworked pool tests

See also [make extensions discoverable by setuptools' entrypoints - #1238](https://github.com/QubesOS/qubes-issues/issues/1238)

### CODE REVIEW

**Comment by kalkin** [comment]
@woju @marmarek
Can we merge this, before our branches diverge again? This is a small atomic change, which is a basis for my other work

File: etc/storage.conf [hunk 1]
```diff
@@ -2,10 +2,11 @@
 driver=xen ; the default xen storage
 ; class = qubes.storage.xen.XenStorage ; class always overwrites the driver
 ; 
-; To use our own storage adapter, you need just to specify the module path and
+; To use our own pool driver it needs to provide `qubes.storage` entry_point
+; see also: http://tinyurl.com/o8czrxg
```
Comment (marmarek) [inline etc/storage.conf]: Can you place here (additionally/instead) a direct link?

Comment (kalkin) [inline etc/storage.conf]: This file will be removed when the pool config is moved to `qubes.xml`.

Comment (marmarek) [inline etc/storage.conf]: Ok.

**Comment by marmarek** [comment]
@woju is quite busy this week, but I think you can assume this will go in in this shape.

File: qubes/storage/__init__.py [hunk 2]
```diff
@@ -465,15 +428,15 @@ def _get_pool_klass(name, config=None):
...
-        klass = load(config.get(name, 'class'))
-    elif config.has_option(name, 'driver'):
-        pool_driver = config.get(name, 'driver')
-        klass = load(qubes.config.defaults['pool_drivers'][pool_driver])
-    else:
-        raise StoragePoolException('Uknown storage pool driver ' + name)
-    return klass
+
+    driver = config.get(name, 'driver')
+    for e in _storage_entry_points():
```
Comment (woju) [inline qubes/storage/__init__.py]: Could you use qubes.get_entry_point_one() function for that? It also takes care of the case when two different packages define one entry point name.

File: etc/storage.conf (hunk 1 above)
Comment (woju) [inline etc/storage.conf]: Still, I would like that not to get into the repository.

**Comment by woju** [comment]
I force-pushed some fixes for those syntax errors. Could you please rebase this pull request onto the new (rebased) branch? Sorry for the problem. (I really shouldn't have committed those syntax errors...)

**Pull request closed without merging**